### ISSUE TITLE
Recording training loss and perplexity during training

### ISSUE THREAD

**Issue by jasonyliang** [huggingface/transformers#7705]
I'm fine-tuning GPT-2 text generation with the following command on Colab:
python run_language_modeling.py \
    --output_dir=$OUTPUT_DIR \
    --model_type=gpt2 \
    --model_name_or_path=$MODEL_NAME \
    --do_train \
    --train_data_file=$TRAIN_FILE \
    --do_eval \
    --eval_data_file=$TEST_FILE \
    --per_gpu_train_batch_size=1 \
    --save_steps=-1 \
    --num_train_epochs=5

I was wondering if I can record the training loss, perplexity ... etc per epoch into a csv file or saved as a variable on Colab?
Thank you so much!
